File: Arrays/balancedStringSplit.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        result = 0
        r = l = 0
        for char in s:
            if char == 'R':
                r += 1
            else:
                l += 1
            if (r == l):
                result += 1
                r = l = 0
        return result

File: Strings/balancedStringSplit.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int result = 0, R = 0, L = 0;
        if (s.size() == 1){
            return 0;
        }
        for (char& c : s){
            if (c == 'R'){
                R++;
            }
            else{
                L++;
            }
            if (R == L){
                result++;
            }
        }
        return result;
    }
};